kernel/src/scan/mod.rs

```diff
@@ -227,14 +227,17 @@ impl Scan {
     }
 
     /// Perform an "all in one" scan. This will use the provided `engine` to read and
-    /// process all the data for the query. Each [`ScanResult`] in the resultant vector encapsulates
+    /// process all the data for the query. Each [`ScanResult`] in the resultant iterator encapsulates
     /// the raw data and an optional boolean vector built from the deletion vector if it was
     /// present. See the documentation for [`ScanResult`] for more details. Generally
     /// connectors/engines will want to use [`Scan::scan_data`] so they can have more control over
     /// the execution of the scan.
     // This calls [`Scan::files`] to get a set of `Add` actions for the scan, and then uses the
```

Comment:
nit: can we update the rest of doc comment to be accurate? L235 to something like 'this calls Scan::scan_data to get an iterator of `ScanData`...